src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/RecordsFetcherFactory.java

```diff
@@ -36,4 +38,6 @@
 
     void setDataFetchingStrategy(DataFetchingStrategy dataFetchingStrategy);
 
+    void setMetricsFactory(IMetricsFactory metricsFactory);
```

Comment:
This isn't needed, instead pass the metrics factory in as part of the create call.  The ShardConumer has access to the metrics factory.